jakarta.ls/src/test/java/org/eclipse/lsp4jakarta/snippets/JakartaSnippetRegistryTest.java

```diff
@@ -0,0 +1,185 @@
+		snippetsContextTest(restGetSnippet, "jakarta.ws.rs.GET", JavaCursorContextKind.BEFORE_METHOD);
+		snippetsContextTest(restPostSnippet, "jakarta.ws.rs.POST", JavaCursorContextKind.BEFORE_METHOD);
+		snippetsContextTest(restPutSnippet, "jakarta.ws.rs.PUT", JavaCursorContextKind.BEFORE_METHOD);
+		snippetsContextTest(restDeleteSnippet, "jakarta.ws.rs.DELETE", JavaCursorContextKind.BEFORE_METHOD);
+		snippetsContextTest(restHeadSnippet, "jakarta.ws.rs.HEAD", JavaCursorContextKind.BEFORE_METHOD);
+				
+	}
+	
+	/**
+	 * Jakarta Servlet snippets - servlet_generic, servlet_doget
+	 * servlet_dopost, servlet_webfilter
+	 */
+	@Test
+	public void ServletSnippetsTest() {
+        Optional<Snippet> servletGenericSnippet = findByPrefix("servlet_generic", registry);
+		assertTrue("Tests has servlet_generic Java snippet", servletGenericSnippet.isPresent());
+		
+		Optional<Snippet> servletDoGetSnippet = findByPrefix("servlet_doget", registry);
+		assertTrue("Tests has servlet_doget Java snippet", servletDoGetSnippet.isPresent());
+		
+		Optional<Snippet> servletDoPostSnippet = findByPrefix("servlet_dopost", registry);
+		assertTrue("Tests has servlet_dopost Java snippet", servletDoPostSnippet.isPresent());
+		
+		Optional<Snippet> servletWebFilterSnippet = findByPrefix("servlet_webfilter", registry);
+		assertTrue("Tests has servlet_webfilter Java snippet", servletWebFilterSnippet.isPresent());
+		
+		snippetsContextTest(servletGenericSnippet, "jakarta.servlet.GenericServlet", JavaCursorContextKind.IN_EMPTY_FILE);
+		snippetsContextTest(servletDoGetSnippet, "jakarta.servlet.http.HttpServlet", JavaCursorContextKind.IN_EMPTY_FILE);
+		snippetsContextTest(servletDoPostSnippet, "jakarta.servlet.http.HttpServlet", JavaCursorContextKind.IN_EMPTY_FILE);
+		snippetsContextTest(servletWebFilterSnippet, "jakarta.servlet.Filter", JavaCursorContextKind.IN_EMPTY_FILE);
+		
+	}
+	
+	
+	/**
+	 * Jakarta Transactions snippets -  tx_user_inject,
+	 * tx_user_jndi, @ Transactional
+	 */
+	@Test
+	public void TransactionsSnippetsTest() {
+        Optional<Snippet> txUserInjectSnippet = findByPrefix("tx_user_inject", registry);
+		assertTrue("Tests has tx_user_inject Java snippet", txUserInjectSnippet.isPresent());
+		
+		Optional<Snippet> txUserJndiSnippet = findByPrefix("tx_user_jndi", registry);
+		assertTrue("Tests has tx_user_jndi Java snippet", txUserJndiSnippet.isPresent());
+		
+		Optional<Snippet> transactionalSnippet = findByPrefix("@Transactional", registry);
+		assertTrue("Tests has @Transactional Java snippet", transactionalSnippet.isPresent());
+		
+		
+		snippetsContextTest(txUserInjectSnippet, "jakarta.transaction.UserTransaction", JavaCursorContextKind.BEFORE_METHOD);
+		snippetsContextTest(txUserJndiSnippet, "jakarta.transaction.UserTransaction", JavaCursorContextKind.BEFORE_METHOD);
+		
+		// getPrefix() method in JavaCursorContextResult is returning empty string instead of prefix. so the below test case is failing.
```

Comment:
In this case the prefix must be "@" in order for the test to pass. Please somehow pass "@" to the `context()` method on line 172. For instance you could test `if (contextType.equals("jakarta.transaction.Transactional")) prefix = "@"` in `snippetsContextTest()` or you could pass a prefix into that test method. Or create a four parameter version of `snippetsContextTest()`.

Reply:
We have made the necessary fixes for the suggested changes. Please review them.